### PR TITLE
Setup Jest

### PR DESCRIPTION
**Pull request by TheJaredWilcurt**
Heard you on Gwen's show tonight. There was talk of setting up testing, which can be a real pain if you haven't done it before. So I made a quick branch.

The primary focus of this PR is just to get testing setup with as few steps/dependencies as possible.

There is one `.js` and one `.vue` file that I wrote tests for as examples, to start off. But I don't fully understand all of the intentions of the original code, so the test expectations may not be accurate.

You can try it with `npm install && npm t`.

I'm mostly just using snapshots on the component tests because they are quick and easy to set up and update, and cover a lot of ground for little effort. In the future if you make an intentional change to a component that causes a snapshot to fail, you can run `npm run test:unit -- -u` to update all snapshots. You can also run `npx majestic` to update one snapshot at a time via a GUI. There's a few tests where I "zoom in" on just the `li` in the wrapper for the snapshot. Zooming in to only snapshot the part of the component relevant to the test is a good practice. It can probably be applied to other tests in this PR. If you're new to snapshot testing, [this should help](https://devchat.tv/views-on-vue/vov-099-testing-in-vue-with-the-jared-wilcurt/).

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "presets": ["@babel/preset-env"]
```
Comment (TheJaredWilcurt) [inline .babelrc:2]: Required for Jest/Babel/Vue to play nice together.

File: src/components/VBreadcrumbs/VBreadcrumbs.vue [hunk 2]
```diff
@@ -38,17 +38,23 @@ export default {
   },
 
   computed: {
-    breadcrumbs() {
+    routeBreadcrumbs() {
```
Comment (TheJaredWilcurt) [inline src/components/VBreadcrumbs/VBreadcrumbs.vue:41]: You cannot have a prop and a computed with the same name (the test wouldn't even load the `.vue` file with that).

So I renamed it and created the next `if` statement. Not sure if this is what the original intention was though.

File: tests/unit/src/components/__snapshots__/VBreadcrumbs.test.js.snap [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      </span>
+    </li>
+  </ol>
+</nav>
+`;
+
+exports[`VBreadcrumbs Crumbs from routes Prevent duplicate child routes (with trailing slash) 1`] = `
+<nav role="navigation" class="vts-breadcrumbs">
+  <ol itemscope="itemscope" itemtype="http://schema.org/BreadcrumbList" class="vts-breadcrumbs__list"></ol>
+</nav>
```
Comment (TheJaredWilcurt) [inline tests/unit/src/components/__snapshots__/VBreadcrumbs.test.js.snap:102]: Not sure if this snapshot or the next properly capture the expected outcome of the scenario. Wasn't sure what the point of the path/parent.path comparison was doing/for.

File: tests/unit/src/utils.test.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+      expect(safeSlot(render, slot))
+        .toEqual(['a'])
+    });
+
+    test('Multiple slots', () => {
+      const slot = ['a', 'b'];
+
+      expect(safeSlot(render, slot))
+        .toEqual('<div>a</div><div>b</div>');
```
Comment (TheJaredWilcurt) [inline tests/unit/src/utils.test.js:83]: Not really sure what the point of `safeSlot` is and why it is different depending on number of slots. So these tests will likely need improved.

File: package.json [hunk 5]
```diff
@@ -54,6 +61,8 @@
     "stylelint": "^13.2.1",
     "stylelint-config-sass-guidelines": "^7.0.0",
     "vue-docgen-api": "^4.14.0",
+    "vue-jest": "^3.0.5",
+    "vue-router": "^3.1.6",
```
Comment (TheJaredWilcurt) [inline package.json:65]: The `package-lock.json` had `vue-router` at a top level until I did a fresh `npm install` for the new testing dependencies. Then it got moved to a lower child level in the package-lock, preventing access to it from all other dependencies. Not sure how it got to the top level to begin with, but anyways, without it apparently the `docs:build` fails, so I just added it as a devDep.

**Comment by Stegosource** [comment]
Hey @TheJaredWilcurt, this is awesome thanks. I am used to using Jest. I think we actually spoke about it on the VoV podcast. Anyway, it's a really good idea to have at least one test in there as an example for anyone else. 

Really funny you chose the breadcrumbs component as that's one of the only ones that isn't actually being used at the moment. I have it in there as one of those copy/paste examples I was talking about. I've got it mostly figured out how I want it to work conceptually, but haven't actually put it together. 

In any case, it is a component which will likely see the light of day in the next release, so the work you've put in is still quite helpful. So thank you.

File: src/components/VBreadcrumbs/VBreadcrumbs.vue (hunk 2 above)
Comment (Stegosource) [reply src/components/VBreadcrumbs/VBreadcrumbs.vue:41]: Yeah, this is mostly the result of a copy/paste job from a different project. This particular component is not being used, but I sometimes work on the DX concepts for how I imagine it being used, then hook things up to work. Haven't gotten to the "hook things up to work" part yet.

File: tests/unit/src/utils.test.js (hunk 4 above)
Comment (Stegosource) [reply tests/unit/src/utils.test.js:83]: `safeSlot` is a simple util I have for checking how many root nodes are inside the slot. Vue only let's us put 1 root node, so if a developer places more than 1, I just wrap them in a `<div>`

**Pull request merged**